javascript/index.js

```diff
@@ -23,18 +23,153 @@
+        getInstruction("mashedPotatoes", 3, (step4) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+          getInstruction("mashedPotatoes", 4, (step5) => {
+            document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
+            document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
+          }, (error) => console.log(error));
+        }, (error) => console.log(error));
+      }, (error) => console.log(error));
+    }, (error) => console.log(error));
+  }, (error) => console.log(error));
+}, (error) => console.log(error))
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction('steak', 0)
+  .then( (step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`
+   return obtainInstruction ('steak', 1)
+  })
+  .then( (step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`
+   return obtainInstruction ('steak', 2)
+  })
+  .then( (step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`
+   return obtainInstruction ('steak', 3)
+  })
+  .then( (step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`
+   return obtainInstruction ('steak', 4)
+  })
+  .then( (step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`
+   return obtainInstruction ('steak', 5)
+  })
+  .then( (step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`
+   return obtainInstruction ('steak', 6)
+  })
+  .then( (step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`
+    document.querySelector("#steak").innerHTML += `<li>${`Steak is ready!`}</li>`
+    document.querySelector("#steakImg").removeAttribute("hidden");
+  
+  })
+ 
+
+
+//Iteration 3 using async/await
+
+async function makeBroccoli() {
+ const step0 = await obtainInstruction('broccoli', 0);
+ document.querySelector("#broccoli").innerHTML += `<li>${step0}</li>`;
+
+
+ const step1 = await obtainInstruction('broccoli', 1);
+ document.querySelector("#broccoli").innerHTML += `<li>${step1}</li>`;
+
+ const step2 = await obtainInstruction('broccoli', 2);
+ document.querySelector("#broccoli").innerHTML += `<li>${step2}</li>`;
+
+ const step3 = await obtainInstruction('broccoli', 3);
+ document.querySelector("#broccoli").innerHTML += `<li>${step3}</li>`;
+
+ const step4 = await obtainInstruction('broccoli', 4);
+ document.querySelector("#broccoli").innerHTML += `<li>${step4}</li>`;
+
+ const step5 = await obtainInstruction('broccoli', 5);
+ document.querySelector("#broccoli").innerHTML += `<li>${step5}</li>`;
+
+ const step6 = await obtainInstruction('broccoli', 6);
+ document.querySelector("#broccoli").innerHTML += `<li>${step6}</li>`;
+
+ const step7 = await obtainInstruction('broccoli', 7);
+ document.querySelector("#broccoli").innerHTML += `<li>${step7}</li>`;
+
+ const step8 = await obtainInstruction('broccoli', 8);
```

Comment:
Here there was no need to create a step8, you could just do what you did on the line below